tests/TestCase/ORM/Association/BelongsToManyTest.php

```diff
@@ -346,6 +346,74 @@ public function testCascadeDeleteDependent()
     }
 
     /**
+     * Test cascading deletes with callback that returns true.
+     *
+     * @return void
+     */
+    public function testCascadeDeleteWithEntityCallback()
+    {
+        $articleTag = $this->getMockBuilder('Cake\ORM\Table')
+            ->setMethods(['deleteAll'])
+            ->getMock();
+        $config = [
+            'sourceTable' => $this->article,
+            'targetTable' => $this->tag,
+            'sort' => ['id' => 'ASC'],
+            'dependent' => function ($tag) {
+                return true;
```

Comment:
Would be good to have an assertion that the `$tag` argument is the entity.

Reply:
Just had a chance to revisit this. I don't think it's done correctly because the entity that I'm actually passing to the callable is the entity that *started* the cascading delete.  But then it makes me wonder which is more useful? A callable on the target entities (that are dependent) or on the source entity?

I honestly ended up not needing this feature for what originally sparked my interest in it.  Do you think it's worth pursuing? I'm happy to do it if you think it would be a good addition, but I'm not sure the best way to proceed.

With deletes, we definitely want zero ambiguity as to what a callable is using as it's argument or bad things could happen.

Reply:
If we don't have a solid set of use cases, I would rather not add features. We can revisit this when we better understand the use cases.

Reply:
Agreed